crates/re_arrow_store/src/store_stats.rs

```diff
@@ -104,56 +108,214 @@ impl DataStore {
+    ///
+    /// This is a best-effort approximation, adequate for most purposes (stats,
+    /// triggering GCs, ...).
+    #[inline]
+    pub fn size_bytes(&self) -> u64 {
+        crate::profile_function!();
+
+        let Self {
+            timeline,
+            cluster_key,
+            inner,
+        } = self;
+
+        (std::mem::size_of_val(timeline) + std::mem::size_of_val(cluster_key)) as u64
+            + inner.read().size_bytes
+    }
+}
+
+impl IndexedBucketInner {
+    /// Computes and caches the size of both the control & component data stored in this bucket,
+    /// in bytes.
+    ///
+    /// This is a best-effort approximation, adequate for most purposes (stats,
+    /// triggering GCs, ...).
+    #[inline]
+    pub fn compute_size_bytes(&mut self) -> u64 {
+        crate::profile_function!();
+
+        let Self {
+            is_sorted,
+            time_range,
+            col_time,
+            col_insert_id,
+            col_row_id,
+            col_num_instances,
+            columns,
+            size_bytes,
+        } = self;
+
+        let control_size_bytes = std::mem::size_of_val(is_sorted)
+            + std::mem::size_of_val(time_range)
+            + std::mem::size_of_val(col_time.as_slice())
+            + std::mem::size_of_val(col_insert_id.as_slice())
+            + std::mem::size_of_val(col_row_id.as_slice())
+            + std::mem::size_of_val(col_num_instances.as_slice())
+            + std::mem::size_of_val(size_bytes);
+
+        let data_size_bytes = compute_columns_size_bytes(columns);
+
+        *size_bytes = control_size_bytes as u64 + data_size_bytes;
+
+        *size_bytes
+    }
+}
+
+// --- Timeless ---
+
+impl PersistentIndexedTable {
+    /// Returns the number of rows stored across this table.
+    #[inline]
+    pub fn total_rows(&self) -> u64 {
+        self.col_num_instances.len() as _
+    }
+
+    /// The size of both the control & component data stored in this table, in bytes.
+    ///
+    /// This is a best-effort approximation, adequate for most purposes (stats,
+    /// triggering GCs, ...).
     #[inline]
     pub fn total_size_bytes(&self) -> u64 {
-        self.inner.read().total_size_bytes
+        let Self {
+            ent_path,
+            cluster_key,
+            col_insert_id,
+            col_row_id,
+            col_num_instances,
+            columns,
+        } = self;
+
+        let control_size_bytes = std::mem::size_of_val(ent_path)
+            + std::mem::size_of_val(cluster_key)
+            + std::mem::size_of_val(col_insert_id.as_slice())
+            + std::mem::size_of_val(col_row_id.as_slice())
+            + std::mem::size_of_val(col_num_instances.as_slice());
+
+        let data_size_bytes = compute_columns_size_bytes(columns);
+
+        control_size_bytes as u64 + data_size_bytes
     }
 }
+
+// --- Common ---
+
+/// Computes the size in bytes of an entire table's worth of arrow data.
+#[allow(dead_code)]
```

Comment:
wrap this and the test below in `#[cfg(tests)] mod tests { … }`  and you won't need `#[allow(dead_code)]`

Reply:
Or inline the function into `compute_table_size_bytes_ignore_headers`